bootstrap.py

```
@@ -0,0 +1,261 @@
+def stop_services(services: list[tuple[str, subprocess.Popen[bytes], IO[str]]]) -> None:
+    info("서비스 종료 중...")
+    for name, proc, log_handle in services:
+        if proc.poll() is None:
+            info(f"종료: {name} (PID={proc.pid})")
+            stop_process_tree(proc.pid)
+        log_handle.close()
+    for port in PORTS:
+        for pid in pids_from_port(port):
+            stop_process_tree(pid)
+    subprocess.run(
+        ["taskkill", "/F", "/IM", "node.exe"],
```

Ruff (0.15.10)

[error] Starting a process with a partial executable path

(S607)